### PR TITLE
Remove the use of SiteConfiguration from FateCommand

### LINKED ISSUE #2155: Remove use of SiteConfiguration in shell's FateCommand
SiteConfiguration is a configuration type intended for server-side code. It is not expected that client-side code will have access to the server-side configuration file. Therefore, it doesn't make sense that the shell, which is a client-side application, should leverage the code in the SiteConfiguration class.

From what I can tell, this is the only remaining non-deprecated use of SiteConfiguration in our client code.

It should be removed.

However, it may not be easy, since the implementation of the command relies on the ability to write directly to ZooKeeper in an area that is intended to be used by server-side utilities. In order to make this command purely client-side, it may need to be written to use RPC calls to a server, so that it does not presume access to the server-side configuration file, handled by the SiteConfiguration class.

(was: https://issues.apache.org/jira/browse/ACCUMULO-2996)

### PR DESCRIPTION
**Pull request by Manno15**
Utilizes thrift to remove any use of SiteConfiguration from `FateCommand`.  Fixes #2155.

WIP as I am still looking into any possible refactors and improvements to the code but any feedback is appreciated. I left a comment inside AdminUtil as I believe additional work can be done there to improve exception handling that coincides with the changes done here to `FateCommand`. There is also a TODO there to remove the use of System.exit.

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (EdColeman) [inline core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:165]: Wondering if it would be better to use char[] or a byte array instead of a string?  Even better would be if the hashed value was used.  If this is mirroring other places in the code, then maybe as a follow on issue as a general improvement in password best practices.

**Comment by EdColeman** [comment]
These are general questions - rather than comment on the code that triggered the question for me, it seems better to just raise them in general.

Would it be possible to replace passing the secret and instead use the authentication of the user that launched the shell?  Depending on granted user permissions, it may be desirable to permit certain FATE command options like print and dump to a wider group than fail or delete.  FATE fail or delete should require "root" access, but other operators would have valid use of print or dump to examine system state. This would enable more restrictive access policies for operations that change system state versus operations that examine the system state on a users granted permissions rather than needing the secret.  One down side of this approach could what is needed to perform the authentication - FATE fail requires the global lock which implies that a manager processes is not going to be available.

If the client authorization cannot be used, wondering if it would be better to use char[] or a byte array instead of a string? Even better would be if the hashed digest value derived from the secret was used in the methods. If this is mirroring other places in the code, then maybe as a follow on issue as a general improvement in password best practices.

**Comment by Manno15** [comment]
@EdColeman. I was following what was already in Fate Command at the time. It allowed the user to pass in a secret instead of using what was in the SiteConfiguration. That is not the default case and only occurs if a user specifies that parameter. I am not quite sure on the initial goal of that though so maybe it is possible to use user authorization but I have not taken a look into it. As for object type, I could replace it with a byte array but the object is only used to create a new `ZooReaderWriter` which takes in a string so I am not sure the gain there.

File: core/src/main/thrift/client.thrift
Comment (ctubbsii) [inline core/src/main/thrift/client.thrift:348]: For a new method that doesn't need to preserve backwards compatibility, the parameters can be numbered in order starting at 1.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:165]: I think I agree with @EdColeman 's observation that you can probably just require these to be used with an authenticated user with Admin permission, rather than provide the instance secret. The server side can use the instance secret from its config file, without needing to pass it.

Not sure from this what values are appropriate for "args". Are these just a list of FaTE transaction IDs to fail? The parameters to these methods could be more specific.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:193]: We wouldn't want TStatus in the public API, but this could be a set of strings for the various statuses. The implementation can convert to the TStatus type, if necessary.

Instead of returning "String" with "Print" in the method name, this could return a map of fate IDs to statuses, or some similar data structure.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:206]: Not sure if we need both "Dump" and "Print" methods. They seem to be doing similar things.

File: core/src/main/java/org/apache/accumulo/fate/AdminUtil.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/fate/AdminUtil.java:545]: If changing these to log messages, including "ERROR" in the message is redundant.

File: core/src/main/java/org/apache/accumulo/fate/AdminUtil.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/fate/AdminUtil.java:558]: To include the stack trace in the log message:

```suggestion
      log.error("Could not read manager lock, not running", e);
```

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:206]: Dump provides a lot more information, I will post a snippet below

<details>

<pre>
root@uno> fate dump
[
  {
    "txid": "58136bf11a1bc7d5",
    "stack": [
      {
        "org.apache.accumulo.manager.tableOps.TraceRepo": {
          "traceId": 0,
          "parentId": 0,
          "repo": {
            "org.apache.accumulo.manager.tableOps.create.CreateTable": {
              "tableInfo": {
                "tableName": "test",
                "namespaceId": {
                  "canonical": "+default"
                },
                "timeType": "MILLIS",
                "user": "root",
                "initialTableState": "ONLINE",
                "initialSplitSize": 19,
                "splitFile": "hdfs://groot:8020/accumulo/tmp/fate-58136bf11a1bc7d5/splits",
                "splitDirsFile": "hdfs://groot:8020/accumulo/tmp/fate-58136bf11a1bc7d5/splitsDirs",
                "props": {
                  "table.majc.compaction.strategy.opts.large.compress.threshold": "100M",
                  "table.iterator.majc.vers": "20,org.apache.accumulo.core.iterators.user.VersioningIterator",
                  "table.constraint.1": "org.apache.accumulo.core.data.constraints.DefaultKeySizeConstraint",
                  "table.iterator.scan.vers.opt.maxVersions": "1",
                  "table.iterator.minc.vers": "20,org.apache.accumulo.core.iterators.user.VersioningIterator",
                  "table.majc.compaction.strategy.opts.large.compress.type": "gz",
                  "table.iterator.majc.vers.opt.maxVersions": "1",
                  "table.majc.compaction.strategy": "org.apache.accumulo.tserver.compaction.strategies.BasicCompactionStrategy",
                  "table.iterator.minc.vers.opt.maxVersions": "1",
                  "table.iterator.scan.vers": "20,org.apache.accumulo.core.iterators.user.VersioningIterator",
                  "table.majc.compaction.strategy.opts.filter.size": "250M"
                }
              }
            }
          }
        }
      }
    ]
  
</pre>

</details>

Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:206]: Whether or not we want all that information I guess is up for debate.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:165]: > Not sure from this what values are appropriate for "args". Are these just a list of FaTE transaction IDs to fail? The parameters to these methods could be more specific.

Yes, Args will be the TiDs. Initially I was going to have it be more broad and include other things inside it. I can see how it is confusing now. Since I have parameters specifically for FilterTiDs for the print statement I could possibly combine the two parameters.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java]: If we had one method that returned the set of transaction IDs, their states, and their blobs like this, code that calls this method can decide what pieces they care about.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java]: Done in my latest commit

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java]: > Instead of returning "String" with "Print" in the method name, this could return a map of fate IDs to statuses, or some similar data structure.

Is there a benefit you had in mind with this possible change? Currently, the returning string is the entire output from `Admin.print` that then gets printed to the shell. Admin print handles all the formatting, the only thing FateCommand has to do is print it to the shell.

Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java]: Instead of 'Print', I could change the name to "fateStatus" as I think that more accurately describes the intended use of the function. It shows the status of either all the txid's in operation or the specific one the user requests.

Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java]: > > Instead of returning "String" with "Print" in the method name, this could return a map of fate IDs to statuses, or some similar data structure.
> 
> Is there a benefit you had in mind with this possible change?

Well, mainly just that APIs should return objects that can be manipulated/read/transformed by calling code. It's not a console REPL where we have only STDIN/STDOUT and have to parse STDOUT from a command. We can have actual typed data. So, there's no reason to resort to "String" types to represent complex information in the API.

> Currently, the returning string is the entire output from `Admin.print` that then gets printed to the shell. Admin print handles all the formatting, the only thing FateCommand has to do is print it to the shell.

Right, that's all that the shell needs to do... but if we're going through the effort of creating a public API endpoint, we're not doing it just for the shell. The shell can take a complex object returned from the public API and format it into a string for printing. But other callers to this API endpoint might want to do something different with the data. When creating a new public API endpoint, we should try to consider the value it has to a generic caller of the public API, and not just the single use case that inspired its creation.

> Instead of 'Print', I could change the name to "fateStatus" as I think that more accurately describes the intended use of the function. It shows the status of either all the txid's in operation or the specific one the user requests.

The name is fine. It could return something that looks like `Set<FateStatus>` or `Map<Txid,Status>` or similar.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:154]: These javadocs should say what the method does normally. The explanation for when an exception is thrown can be in a javadoc `@throws SomeException when <explanation here>` line.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:191]: See previous review conversation for discussion about consolidating these, and having a meaningful return type other than String. These APIs should not be written exclusively for use with the Shell, but should have meaningful return types of their own, and not just formatted strings for the shell.

File: core/src/main/thrift/client.thrift
Comment (ctubbsii) [inline core/src/main/thrift/client.thrift:342]: This can be numbered starting at 1. The numbers for the throws exceptions can also start at 1. They are a separate context, and it doesn't matter if they reuse numbers across contexts, as long as the numbers aren't reused within the same parameter/exception context.

**Review by ctubbsii — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java]: Okay, that makes a lot of sense. My viewpoint was too narrowly focused.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:154]: This was just a mistake by me. I pushed those up without editing the description of them.

**Comment by Manno15** [comment]
Converting to draft PR as the most recent changes are still a WIP. I would still appreciate some feedback. Specifically with moving `TransactionStatus` to public API.

**Comment by Manno15** [comment]
I have finalized the consolidation of the fate print and dump options. Now, the stack info gets stored inside `TransactionStatus` as well. The only part I am unsure of is creating the gson object inside adminUtil but I couldn't find any alternatives.

**Comment by Manno15** [comment]
The latest commit includes the upgrade to thrift 0.15.0

**Comment by milleruntime** [comment]
These changes look pretty good and I think this would be nice to get into 2.1. @Manno15 what kind of testing were you able to do? My only concern is leaking the FATE types in `TransactionStatus`. We do blind casting of the serialized FATE repos on the server side so I think it might be possible to return some of the types not in the API.

**Comment by Manno15** [comment]
It has been awhile, so I do not fully recall. I will look into testing it
again.

**Comment by milleruntime** [comment]
While testing with Uno and the `SlowIterator`, I noticed that multiple FATE Txs get kicked off if you issue multiple compact commands. This may be unrelated to your changes but I think it is noteworthy. I thought there were checks that prevent another compaction from running if the table doesn't need to be compacted. It could be that the check for this happens in CompactRange, after the FATE Tx gets started.
<pre>
root@uno> fate print
txid: 1bc091ffb9162774  status: IN_PROGRESS         op: CompactRange     locked: [R:+default, R:1] locking: []              top: CompactionDriver created: 2022-03-04T16:38:44.088Z
txid: 543834dd76654bc1  status: IN_PROGRESS         op: CompactRange     locked: [R:+default, R:1] locking: []              top: CompactionDriver created: 2022-03-04T16:56:29.574Z
txid: 7e20a7508cac33d9  status: IN_PROGRESS         op: CompactRange     locked: [R:+default, R:1] locking: []              top: CompactionDriver created: 2022-03-04T16:57:25.536Z
txid: 1ca0ac1b2198cfe0  status: IN_PROGRESS         op: CompactRange     locked: [R:+default, R:1] locking: []              top: CompactionDriver created: 2022-03-04T16:38:17.825Z
 4 transactions
root@uno> listcompactions
 SERVER               | AGE       | TYPE  | REASON | READ  | WROTE | TABLE           | TABLET                                   | INPUT | OUTPUT                              | ITERATORS | ITERATOR OPTIONS
ip-10-113-12-25:10000 |    19m15s |  FULL |   USER |  224K |  224K |              ci | 1<<                                      |     3 |   /1/default_tablet/A000005e.rf_tmp |        [] | {}
</pre>

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (milleruntime) [inline core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:183]: I think these are good so far but we need a way to get the `TxIds`. We need a method like this on InstanceOperations:
`List<TransactionStatus> fatePrint() throws AccumuloException;`

Here is an example of how it could be used:
<pre>
var list = client.instanceOperations().fatePrint();
for (var ts : list) {
  client.instanceOperations().fateFail(List.of(ts.getTxid()));
}
</pre>

I also think this would be a good opportunity to use a stronger type for Txid. This was discussed recently here: https://github.com/apache/accumulo/issues/2473#issuecomment-1039473784

**Review by milleruntime — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java:183]: If this would benefit from some of the more aggressive FaTE refactoring you're working on that might go in 3.0, I'd be okay with pushing this PR off until then. But, otherwise, I'd generally agree with your comment above that this would be good to get into 2.1 (based on the intended outcome, not the code itself, since I haven't yet had a chance to review the code).

**Comment by Manno15** [comment]
The latest commit fixed the new merge conflicts. This PR is now more a WIP. Several changes to FateCommand has happened and this will need to be further tested to make sure everything still works. Also, the `FateCommandTest` needs further changes due to the API changes and removal of `SiteConfiguration` from FateCommand. It might make sense for the test to be moved to an IT, such as `FateIT`.  Thoughts on the test @milleruntime?

**Comment by Manno15** [comment]
The build errors are because I have parts of the test commented out for the moment.

**Comment by milleruntime** [comment]
> The latest commit fixed the new merge conflicts. This PR is now more a WIP. Several changes to FateCommand has happened and this will need to be further tested to make sure everything still works. Also, the `FateCommandTest` needs further changes due to the API changes and removal of `SiteConfiguration` from FateCommand. It might make sense for the test to be moved to an IT. Thoughts on the test @milleruntime?

Possibly. But I just created `FateCommandTest` to have as a unit test since we had nothing before. I have more changes with my refactor in https://github.com/apache/accumulo/pull/2475

**Comment by Manno15** [comment]
@dlmarion I merged in your changes in #2575. Let me know I missed anything. I am still working on adapting the tests from the previous changes as well.

**Comment by Manno15** [comment]
This should mostly be ready for review. I confirmed fail, delete, list, print, and dump work as expected. I am not quite getting cancele-submitted to work at the moment.

**Review by milleruntime — APPROVED** [review]
I think this looks good. @ctubbsii did you want to give it another look before merging?

**Comment by ctubbsii** [comment]
> I think this looks good. @ctubbsii did you want to give it another look before merging?

I would like to take a look, but I might not have time until next week.

**Comment by milleruntime** [comment]
@Manno15 Looks like some recent thrift refactoring caused a merge conflict in FateCommand with these changes. Any chance you can resolve them in your branch?

File: core/src/main/java/org/apache/accumulo/fate/AdminUtil.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/fate/AdminUtil.java:291]: What are these type adapters doing? An inline comment could help.

File: core/src/main/java/org/apache/accumulo/fate/AdminUtil.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/fate/AdminUtil.java:422]: These are a bit of a change in behavior for the shell, which I think currently uses AdminUtil. I think this behavior is better, but might want to include some release notes that mention the change, so if anybody was relying on scripting the shell, and capturing STDOUT, it will work differently for them, depending on their log configuration.

File: core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java:321]: It seems like AccumuloSecurityException should propagate up.

File: server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java
Comment (ctubbsii) [inline server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java:468]: In addition to authenticating, we probably want to restrict this to users with admin permissions, right?

**Review by ctubbsii — CHANGES_REQUESTED** [review]

**Comment by Manno15** [comment]
@ctubbsii "What are these type adapters doing? An inline comment could
help."

The type adapters were initially in the 'FateCommand' file and moved into
'AdminUtil' (along with the code that used it) to help pretty print the
json stack in particular.

**Comment by milleruntime** [comment]
@ctubbsii any of your request blocking this merge? If not, I will merge this and do a follow up.

**Comment by ctubbsii** [comment]
> @ctubbsii any of your request blocking this merge? If not, I will merge this and do a follow up.

Well, I think the permissions issue should be addressed before this is merged.

File: server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java
Comment (DomGarguilo) [reply server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java:468]: What do you mean by "admin permissions" here? It looks like this checks that the credentials contain `SYSTEM` permissions.

Comment (ctubbsii) [reply server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java:468]: > What do you mean by "admin permissions" here? It looks like this checks that the credentials contain `SYSTEM` permissions.

What line do you see that?

Comment (ctubbsii) [reply server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java:468]: I mean `SystemPermission.SYSTEM`, or `security.canPerformSystemActions(credentials)`

**Comment by Manno15** [comment]
I believe `authenticate` checks for `SystemPermission.SYSTEM.

**Comment by ctubbsii** [comment]
> I believe `authenticate` checks for `SystemPermission.SYSTEM.
> 

No, it does not. The API for authenticating allows system users to authenticate a provided set of user credentials to be authenticated, but it does not check that the user being authenticated has the system permission, unless the user is attempting to authenticate themselves, but in that case, the system permission is only being checked for the very limited purpose of authenticating... not for checking whether a user can perform a general system action.

File: core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java
Comment (DomGarguilo) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java:321]: @ctubbsii, how far up do you think it should propagate? Should it be caught and handled where `executeAdminOperation` is called, or propagate up even further than that?

Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java:321]: AccumuloSecurityException is typically intended to be thrown from our public API, so it should propagate all the way to that. The shell is probably already set up to catch it and log an error message/stack trace if any commands throw it.

**Review by ctubbsii — CHANGES_REQUESTED** [review]
Consider using Set instead of List when the items are expected to be unique.

We need a way to get the list of transaction IDs to be able to use these methods in the API.

Avoid using "print" for the API/RPC operations. "Printing" is a feature of the shell/console, and is something that is done with the results retrieved from the API. "listing" or "getting" are more appropriate verbs for the API/RPC operations.

I believe FateAdmin currently allows offline manipulation of the fate transactions. If so, we need to be careful to preserve that behavior when we add this public API to do it via the shell. We need to not remove functionality that could help perform "surgery" on an offline system in a bad state.

**Comment by milleruntime** [comment]
In my attempts to try and create a test for #2754 I added this test to `ShellServerIT`:
<pre>
@Test
  public void testFateCommandWithSlowCompaction() throws Exception {
    final String table = getUniqueNames(1)[0];

    // compact
    ts.exec("createtable " + table);

    String tableId = getTableId(table);

    // make two files
    ts.exec("insert a b c d");
    ts.exec("flush -w");
    ts.exec("insert x y z v");
    ts.exec("flush -w");
    int oldCount = countFiles(tableId);

    // setup SlowIterator to sleep for 10 seconds
    ts.exec("config -t " + table + " -s table.iterator.majc.slow=1,org.apache.accumulo.test.functional.SlowIterator");
    ts.exec("config -t " + table + " -s table.iterator.majc.slow.opt.sleepTime=10000");

    // merge two files into one
    ts.exec("compact -t " + table);

    // compaction should still be running
    //assertFalse(countFiles(tableId) < oldCount);

    log.info("Calling fate print for table = {}", table);
    ts.exec("fate -print", true, "txid:", true);

    ts.exec("deletetable -f " + table);
  }
</pre>

This test will usually fail on main but on this branch it passes.

File: core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java
Comment (DomGarguilo) [reply core/src/main/java/org/apache/accumulo/core/client/admin/InstanceOperations.java]: Addressed in [fb3de50](https://github.com/apache/accumulo/pull/2215/commits/fb3de50f2fceb48ed142ab0f8aabd05e180670ab)

**Comment by EdColeman** [comment]
I believe that additional test from @milleruntime fails on current main because the fate command wants access to the accumulo.properties file.  When mini sets up, it uses the accumulo.properties in a mini /conf directory and the properties file is not on the classpath.  Normally the shell IT test is using accumulo-client.properties so not having the accumulo.properties file is **"normal"**.  

The changes in this branch remove the need for accumulo.properties - and that's why it passes here, but fails on current (7f42c7d72a) main.  The test will pass (with other logic changes) on main if the test is modified so that accumulo.properties is set for the FATE command with:
```
    System.setProperty("accumulo.properties",  "file://" + getCluster().getConfig().getAccumuloPropsFile().getCanonicalPath());
```

I added the test to #2755, and with the logic change, the new test will pass. Once this PR is merged, the test should be modified so that only the client properties file is necessary to pass

File: core/src/main/java/org/apache/accumulo/core/client/admin/TransactionStatus.java
Comment (DomGarguilo) [reply core/src/main/java/org/apache/accumulo/core/client/admin/TransactionStatus.java]: Addressed in [e51f18b](https://github.com/apache/accumulo/pull/2215/commits/e51f18b3d638a04dce75afa3d6c37ab2a66a9c8c)

**Comment by DomGarguilo** [comment]
> Well, I think the permissions issue should be addressed before this is merged.

Attempted to address this in [bf2b32b](https://github.com/apache/accumulo/pull/2215/commits/bf2b32b920a15e25610784323688b8ba83aab4e3)

**Comment by milleruntime** [comment]
Closing this in favor of #2780

**Pull request closed without merging**
